modules/apps/frontend-js/frontend-js-components-web/src/main/resources/META-INF/resources/autocomplete/Autocomplete.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayAutocomplete from '@clayui/autocomplete';
+import ClayDropDown from '@clayui/drop-down';
+import {FocusScope} from '@clayui/shared';
+import React, {useEffect, useRef, useState} from 'react';
+
+const DefaultItemsWrapperRenderer = ({
+	items,
+	labelKey = 'label',
+	onItemClick,
+	valueKey = 'value',
+}) => {
+	return (
+		<ClayDropDown.ItemList className="mb-0">
+			{items && items.length === 0 && (
+				<ClayDropDown.Item className="disabled">
+					{Liferay.Language.get('no-items-were-found')}
+				</ClayDropDown.Item>
+			)}
+			{items &&
+				items.length > 0 &&
+				items.map((item) => (
+					<ClayAutocomplete.Item
+						key={String(item[valueKey])}
+						onClick={() => onItemClick(item)}
+						value={String(item[labelKey])}
+					/>
+				))}
+		</ClayDropDown.ItemList>
+	);
+};
+
+function Autocomplete({
```

Comment:
Hey @bryceosterhaus looking at this well maybe we can support this component in Clay as a high-level component, it looks like you've eliminated a lot of commerce dependencies here.

Reply:
Yes I agree, it doesn't need to be done now, we can check it out later and we can take care of that too.